ts/src/test/test.ts

```diff
@@ -587,25 +606,29 @@ export default class testMainClass extends baseMainTestClass {
         if (!this.privateTestOnly) {
             if (exchange.has['spot'] && spotSymbol !== undefined) {
                 if (this.info) {
-                    dump ('[INFO:SPOT TESTS]');
+                    dump ('[INFO:SPOT TESTS]', this.exchangeTitle (exchange));
                 }
                 exchange.options['type'] = 'spot';
+                exchange.options['defaultType'] = 'spot';
                 await this.runPublicTests (exchange, spotSymbol);
             }
             if (exchange.has['swap'] && swapSymbol !== undefined) {
                 if (this.info) {
-                    dump ('[INFO:SWAP TESTS]');
+                    dump ('[INFO:SWAP TESTS]', this.exchangeTitle (exchange));
                 }
                 exchange.options['type'] = 'swap';
+                exchange.options['defaultType'] = 'swap';
                 await this.runPublicTests (exchange, swapSymbol);
             }
         }
         if (this.privateTest || this.privateTestOnly) {
             if (exchange.has['spot'] && spotSymbol !== undefined) {
+                exchange.options['type'] = 'spot';
```

Comment:
what is this `type` ?

Reply:
I thought to add that because of this : https://github.com/ccxt/ccxt/blob/97a87c51d8667e03ef21840f26f4c51582f0de94/ts/src/base/Exchange.ts#L3424 
but yeah, if we set `defaultType` , then no need to set `type`.
removed it now, and tests passed.

Reply:
This PR ready too.